### PR TITLE
JWT support for OAuth2 access tokens

### PR DESCRIPTION
**Pull request by evert**
Defined here: https://tools.ietf.org/html/draft-ietf-oauth-access-token-jwt-12

If JWT_PRIVATE_KEY is set and contains a RS256 private key, access
tokens will be generated as JWTs.

More related features to this will follow, but this is the first step to
JWT support.

### CODE REVIEW

File: src/oauth2/controller/active-sessions.ts [hunk 1]
```diff
@@ -38,4 +44,11 @@ class UserActiveSessions extends Controller {
 
 }
 
+function assertNotGroup(input: Principal): input is User | App {
```
Comment (juhangsin) [inline src/oauth2/controller/active-sessions.ts:47]:
```suggestion
function assertNoneGroup(input: Principal): input is User | App {
```

wouldn't this make more sense?

File: src/oauth2/controller/active-sessions.ts [hunk 2]
```diff
@@ -20,6 +22,10 @@ class UserActiveSessions extends Controller {
       throw new Forbidden('You can only use this API for yourself yourself, or if you have \'admin\' privileges');
     }
 
+    if (!assertNotGroup(user)) {
```
Comment (juhangsin) [inline src/oauth2/controller/active-sessions.ts:25]:
```suggestion
    if (!assertNoneGroup(user)) {
```
Wouldn't this make more sense?

**Review by juhangsin — APPROVED** [review]
just a name change (i think), but otherwise, looks great!

File: src/oauth2/controller/active-sessions.ts (hunk 1 above)
Comment (evert) [reply src/oauth2/controller/active-sessions.ts:47]: It asserts that 'input' is _not_ a group. Input can be either a user, group or app but it's always just 1.

An alternative could have been `assertsAppOrUser` but it's a bit more clumsy to write. Or `assertsPrincipalHasOAuthSupport` but it's getting a bit wordy then :/
